dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/audit/DefaultTrackedEntityAuditServiceTest.java

```diff
@@ -0,0 +1,115 @@
+ */
+package org.hisp.dhis.tracker.audit;
+
+import static org.hisp.dhis.audit.AuditOperationType.READ;
+import static org.hisp.dhis.test.utils.Assertions.assertHasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.List;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.feedback.BadRequestException;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.test.integration.PostgresIntegrationTestBase;
+import org.hisp.dhis.trackedentity.TrackedEntityAudit;
+import org.hisp.dhis.trackedentity.TrackedEntityAuditQueryParams;
+import org.hisp.dhis.tracker.TestSetup;
+import org.hisp.dhis.tracker.export.enrollment.EnrollmentOperationParams;
+import org.hisp.dhis.tracker.export.enrollment.EnrollmentService;
+import org.hisp.dhis.user.CurrentUserUtil;
+import org.hisp.dhis.user.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+@TestInstance(Lifecycle.PER_CLASS)
+@Transactional
+class DefaultTrackedEntityAuditServiceTest extends PostgresIntegrationTestBase {
+  @Autowired private EnrollmentService enrollmentService;
+  @Autowired private TrackedEntityAuditService auditService;
+  @Autowired private TestSetup testSetup;
+
+  private final UID program = UID.of("BFcipDERJnf");
+  private final UID trackedEntity = UID.of("QS6w44flWAf");
+  private TrackedEntityAuditQueryParams params;
+  private int countBeforeTest;
+
+  @BeforeAll
+  void setUp() throws IOException {
+    testSetup.importMetadata();
+
+    User importUser = userService.getUser("tTgjgobT1oS");
+    injectSecurityContextUser(importUser);
+
+    testSetup.importTrackerData();
+
+    params = new TrackedEntityAuditQueryParams();
+    params.setTrackedEntities(List.of(trackedEntity.getValue()));
+    params.setAuditTypes(List.of(READ));
+  }
+
+  @BeforeEach
+  void beforeEach() {
+    countBeforeTest = auditService.getTrackedEntityAuditsCount(params);
+  }
+
+  @Test
+  void shouldCreateReadAuditWhenGettingEnrollmentAndTrackedEntitySupplied()
+      throws ForbiddenException, BadRequestException {
+    enrollmentService.findEnrollments(
+        EnrollmentOperationParams.builder().program(program).trackedEntity(trackedEntity).build());
+    List<TrackedEntityAudit> actualAudits = auditService.getTrackedEntityAudits(params);
+
+    assertHasSize(countBeforeTest + 1, actualAudits);
+    TrackedEntityAudit expectedAudit =
+        new TrackedEntityAudit(
+            trackedEntity.getValue(), CurrentUserUtil.getCurrentUsername(), READ);
+    assertAudit(expectedAudit, actualAudits.get(0));
+  }
+
+  @Test
+  void shouldNotCreateReadAuditWhenGettingEnrollmentAndTrackedEntityNotSupplied()
```

Comment:
We could add the missing test case `shouldNotCreateReadAuditWhenNotGettingEnrollmentAndTrackedEntitySupplied`

Reply:
what do you mean by `NotGettingEnrollment`? enrollment not found?